espnet2/gan_codec/shared/decoder/seanet.py

```diff
@@ -0,0 +1,252 @@
+# Adapted from https://github.com/facebookresearch/encodec
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+
+"""Encodec SEANet-based encoder and decoder implementation."""
+
+from typing import Any, Dict, List, Optional, Tuple
+
+import numpy as np
+import torch
+from torch import nn
+from torch.nn import functional as F
+from torch.nn.utils import spectral_norm, weight_norm
+
+from espnet2.gan_codec.shared.encoder.seanet import (
+    SLSTM,
+    SConv1d,
+    SEANetResnetBlock,
+    apply_parametrization_norm,
+    get_norm_module,
+)
+
+
+def unpad1d(x: torch.Tensor, paddings: Tuple[int, int]):
+    """Remove padding from x, handling properly zero padding. Only for 1d!"""
+    padding_left, padding_right = paddings
+    assert padding_left >= 0 and padding_right >= 0, (padding_left, padding_right)
+    assert (padding_left + padding_right) <= x.shape[-1]
+    end = x.shape[-1] - padding_right
+    return x[..., padding_left:end]
+
+
+class NormConvTranspose1d(nn.Module):
```

Comment:
maybe these layers could be gathered to a separate folder? Such as "espnet2/codec/layers"?

Reply:
Thanks for the suggestion! I generally agree with your point on setting a better organization of the modules. But I'm not pretty sure if the above layers are good to be separated. Mostly because they are not beyond the original torch function but just wrappers for the seanet module itself.

Reply:
Please let me know if you think layers would be still better in the case~